### DISCUSSION TITLE
How to read container's stdout when using passthough log driver?

### DISCUSSION THREAD

**Discussion by tkit1994** [containers/podman#17348] [General]
podman-kube@.service has switched to passthough log driver by default in podman version 4.4.0.
How can it read a container's stdout.
```bash
➜  journal podman logs alist-alist
Error: this container is using the 'passthrough' log driver, cannot read logs: this container is not logging output
```
```bash
➜  journal podman attach alist-alist
Error: attaching to container f9a23ad068936cee9e8a85e529fa35bc67717b7de1f6094265cc2c83abd62554: this container is using the 'passthrough' log driver, cannot attach: this container is not logging output
```

**Comment by vrothberg** [comment]
Thanks for reaching out, @tkit1994!  Is this change breaking your deployment?

With `passthrough`, you can find the logs in the journal.  You can use `journalctl _PID=$PID_of_process` for instance.

**Reply by tkit1994** [reply]
Thanks, it does not.  Your solution works like a charm.

**Reply by vrothberg** [reply]
Excellent.  Thanks a lot for reaching out!

**Comment by Luap99** [comment]
I was thinking maybe we can teach `podman logs` to still read from the journal even with `passthrough` when it is run in a systemd unit, we already set `PODMAN_SYSTEMD_UNIT` unit env so we can use the unit name to filter logs. Most of the code to read the journal correctly is already there, we just need to apply different filters and slightly change the message parsing logic.

**Reply by vrothberg** [reply]
Yes, that will work :+1:

**Comment by rhatdan** [comment]
That sounds like a great idea.

**Comment by rrbanda** [comment]
@rhatdan is this specific to OS on which podman runs ? 
Because i can see logs on a rocky linux for the same container but not on a CentOS 9

**Comment by rhatdan** [comment]
No there should be no diference on OS, but their could be difference on versions or how you have containers.conf configured between the two machines.
